packages/hardhat-etherscan/src/solc/bytecode.ts

```diff
@@ -39,6 +40,7 @@ export interface ContractInformation extends BytecodeExtractedData {
   sourceName: SourceName;
   contractName: ContractName;
   contract: CompilerOutput["contracts"][SourceName][ContractName];
+  undetectableLibraries?: LibraryNames;
```

Comment:
This, and the use of `ContractInformation` in some places and `Required<ContractInformation>` in others, seems like a code smell to me. I don't know how to improve it though, probably some re-design is needed.

Reply:
Indeed. The `ContractInformation` type is a bit ad hoc. However, since it is not obvious how to address it, we might want to write it down in a comment or open up an issue here in github. I think opening an issue is overkill for something like this though.